teletext-twitter/output.py

```diff
@@ -86,19 +102,34 @@ def write_tweets(twitter_object, mode, count, config, query=None): # grab the la
         tweet_human_time = time.strftime("%d-%b-%Y %H:%S", tweet_time) # reformat time/date output
         tweet_username = charsub(status.user.screen_name)
 
-        with open(filename, "a") as file:
+        with open(filename, "a", newline="") as file:
             post_length = len(tweet_text) + 1 # how long is our next tweet? (including info line)
-            if (line_position + post_length) > 25: # are we about to go over the page?
-                for blankline_position in range(line_position, 25): # how many blank lines do we need?
-                    file.write("OL,{},".format(blankline_position) + (" " * 15) + "\r\n")
+            if (line_position + post_length) > 24: # are we about to go over the page?
                 file.write("FL,0,0,0,0,0,100\r\n")
+                if subpage_enhancements:
+                    write_ehancements(file, subpage_enhancements)
                 subpage += 1 # start a new page
+                subpage_enhancements = []
                 if subpage > 99:
-                    break # reached subpage limit - dump the rest
+                    return # reached subpage limit - dump the rest
                 write_header(file, subpage, max_subpages, config)
                 line_position = 4 # and reset our cursor
-            write_tweet_info(file, line_position, tweet_username, tweet_human_time, config)
+            tweet_username_enhanced = charenhance(tweet_username,(37-len(tweet_human_time)-len(tweet_username)))
+            if tweet_username_enhanced[1]:
+                subpage_enhancements.append([line_position+40,4,0]) # active position to start of row
+                subpage_enhancements += tweet_username_enhanced[1]
+            write_tweet_info(file, line_position, tweet_username_enhanced[0], tweet_human_time, config)
             line_position += 1
             for line in tweet_text:
-                write_tweet_line(file, line_position, line, config)
+                tweet_text_line = charenhance(line,1)
+                if tweet_text_line[1]:
+                    subpage_enhancements.append([line_position+40,4,0]) # active position to start of row
+                    subpage_enhancements += tweet_text_line[1]
+                write_tweet_line(file, line_position, tweet_text_line[0], config)
                 line_position += 1
+    
+    # finish the last subpage
+    with open(filename, "a", newline="") as file:
+        file.write("FL,0,0,0,0,0,100\r\n")
+        if subpage_enhancements:
+            write_ehancements(file, subpage_enhancements)
```

Comment:
Typo here too. Also this is great I didn’t think of finishing pages.

Reply:
I'm possibly being blind, can you explain what you think is incorrect with these?

Reply:
oh, now I see, yes ehancements should of course be enhancements everywhere :)